examples/salesforce/index.js

```diff
@@ -0,0 +1,108 @@
+
+const app = express();
+const port = 3000;
+const serviceId = 'sfdc';
+const userId = 'usr-123'; // normally determined through authorization
+const tenantId = 'contoso'; // organization/project userId is part of
+
+app.get('/', (req, res) => {
+  res.redirect('/integrations');
+});
+
+// The management page that shows connected services
+app.get(`/integrations`, async (req, res) => {
+  // Check if the user's organization has already authorized access to SFDC
+  const credentials = await everyauth.getIdentity(serviceId, { userId, tenantId }); // TODO: this should only be tenantId
+  if (credentials) {
+    // Present options to reconnect or disconnect, and test
+    res.send(`
+            <h2>Salesforce</h2>
+            <p>Connected to ${credentials.native.instance_url}</p>
+            <form action="/integrations/sfdc/connect">
+                <input type="submit" value="Reconnect" />
+            </form>
+            <form action="/integrations/sfdc/disconnect">
+                <input type="submit" value="Disconnect" />
+            </form>
+            <form action="/newsletter/signup">
+                <input type="submit" value="Test newsletter signup" />
+            </form>
+        `);
+  } else {
+    // Present option to connect
+    res.send(`
+            <h2>Salesforce</h2>
+            <p>Service not connected</p>
+            <form action="/integrations/sfdc/connect">
+                <input type="submit" value="Connect" />
+            </form>
+        `);
+  }
+});
+
+// Endpoint that initiates authorization process to Salesforce
+app.use(
+  '/integrations/sfdc/connect',
+  everyauth.authorize(serviceId, {
+    finishedUrl: '/integrations/sfdc/connected',
+    mapToTenantId: (req) => tenantId,
+    mapToUserId: (req) => userId,
+  })
+);
+
+// Endpoint redirected to when Salesforce authorization process has completed
+app.get('/integrations/sfdc/connected', async (req, res) => {
+  if (req.query.error) {
+    res.send(`<p>There was an error authorizing access to SFDC: ${req.query.error}<p>`);
+  } else {
+    res.redirect('/integrations');
+  }
+});
+
+// Endpoint used to disconnect from a service
+app.get('/integrations/sfdc/disconnect', async (req, res) => {
+  const identities = await everyauth.getIdentities(serviceId, { userId, tenantId }); // TODO: this should only be tenantId
+  for (const identity of identities.items) {
+    await everyauth.deleteIdentity(serviceId, identity.id);
+  }
+  res.redirect('/integrations');
+});
+
+// Test page that contains a newsletter signup form and postback logic
+app.get('/newsletter/signup', async (req, res) => {
+  if (req.query.email) {
+    // Process form submission
+    const credentials = await everyauth.getIdentity(serviceId, { userId, tenantId }); // TODO: this should only be tenantId
+    if (credentials) {
+      // Salesforce is connected, create a Contact record
+      const sfdcClient = new jsforce.Connection({
+        instanceUrl: credentials.native.instance_url,
```

Comment:
Add a little bit more of context what's the instance_url in the context of Salesforce.

Reply:
The blog will go over it in more detail.